llvm/utils/lit/tests/Inputs/shtest-glob/glob-echo.txt

```diff
@@ -0,0 +1,2 @@
+## Tests glob pattern expansion by listing matching files.
+# RUN: echo %{inputs}/shtest-glob/example_file*.txt
```

Comment:
Thanks, this is good. Can we also try testing a glob expansion passes to a non-builtin command? E.g. a local python script?

Reply:
Could also do something like `touch %t1.txt %t2.txt` and then try `mkdir %t*.txt` and check that we get two file exists errors.

Reply:
Thank you for the suggestions! I have a couple of follow-up questions to ensure I'm on the right track.

For testing glob expansion with a non-built-in command, would it be appropriate to use the `find` command? I was considering using `find` to list files that match a glob pattern. Do you think this would effectively demonstrate the glob expansion, or is there another command you would recommend?

Regarding the mkdir suggestion, I'm a bit confused about where this test should be placed. Should it go in the same file as the glob-echo test (`llvm/utils/lit/tests/Inputs/shtest-glob/glob-echo.txt`), or would it be better to create a new file specifically for mkdir, such as `llvm/utils/lit/tests/Inputs/shtest-glob/glob-mkdir.txt`? And then, in `llvm/utils/lit/tests/shtest-glob.py`, should this test be expected to fail, indicating the current behavior before implementing the fix?

Reply:
I'm not too familiar with the structure of the lit internal test suite - I'd say keeping the tests in separate files matches the current layout more closely. Looking at the code mkdir handles globs already so if you just run something like `not mkdir %S/example_input%.input` you can verify the error output.